lib/ads/client_side_ad_manager.js

```diff
@@ -84,8 +84,11 @@ shaka.ads.ClientSideAdManager = class {
    * @param {!google.ima.AdsRequest} imaRequest
    */
   requestAds(imaRequest) {
-    goog.asserts.assert(imaRequest.adTagUrl.length,
-        'The ad tag needs to be set up before requesting ads.');
+    goog.asserts.assert(
+        (imaRequest.adTagUrl && imaRequest.adTagUrl.length) ||
+        (imaRequest.adsResponse && imaRequest.adsResponse.length),
```

Comment:
The empty string is false, so this could just be `imaRequest.adTagUrl || imaRequest.adsResponse`.

Reply:
Done